evalml/automl/automl_search.py

```diff
@@ -575,7 +588,15 @@ def _find_best_pipeline(self):
             if self._train_best_pipeline:
                 X_threshold_tuning = None
                 y_threshold_tuning = None
-                X_train, y_train = self.X_train, self.y_train
+                if self._best_pipeline.model_family == ModelFamily.ENSEMBLE:
+                    X_train, y_train = self.X_ensemble, self.y_ensemble
+                else:
+                    if self.X_ensemble:
+                        X_train = _convert_woodwork_types_wrapper(self.X_train.to_dataframe()).append(_convert_woodwork_types_wrapper(self.X_ensemble.to_dataframe()), ignore_index=True)
```

Comment:
@bchen1116 I think we need something like this:

In AutoMLSearch init
```python
self.X_train = infer_feature_types(X_train)
# This part I'm not so sure about
self.X_feature_types = self.X_train.logical_types
```

In _find_best_pipeline:
```python
X_train = infer_feature_types(self.X_train.to_dataframe().append(self.X_ensemble.to_dataframe()), self.X_feature_types)
```

The issue is that we want to preserve the types provided by the user in case they override the default types. 

Maybe @angela97lin knows the best way/better way to concatenate two datatables while preserving types?

Reply:
I think what you've proposed looks good here @freddyaboulton! Unfortunately, I'm not sure there's a good way to concatenate two DataTables without going to the underlying DataFrames :'(

Reply:
Got it! I'll put that in, thanks for the info!